src/schoology-extractor/schoology_extractor/schoology_request_service.py

```diff
@@ -0,0 +1,112 @@
+# SPDX-License-Identifier: Apache-2.0

```

Comment:
I am not sure if schoology_request_service is a good name for this file... Visitors might infer from the name, that they should look into this file for making requests to the Schoology API, while this is actually both getting data from the API(but not directly exposing the get and other request methods inside of request_client) and exporting it to csv, to then be called by the main script. 
I would also add docstrings to this class.

Reply:
I don't particularly like the name but have not been able to come up with anything better. I have not found a good, consistent pattern name that applies to the main business logic of a back-end component. At first I thought of facade:

> "Provide a unified interface to a set of interfaces in a subsystem. Facade defines a higher-level interface that makes the subsystem easier to use."

That is close, but in my experience a facade usually sits over one service or a couple of very similar services that belong together. In this case we are orchestrating the calls to the API service and to the filesystem.

Orchestrator is a possibility. However, I usually think of the `main` script more with that term. 

I used the word "service" when about this being a _service layer_. Maybe `schoology_extractor_service` would be better?

Reply:
Yeah, schoology_extractor_service sounds better, but I'd add docstring to that class

Reply:
After looking at this more closely again, I realized I was getting myself confused: it is not exporting to CSV. It is just getting data and reshaping. So "extractor" is not good in the name. "Facade" comes back into favor then, since a Facade will often reshape a Data Transfer Object to a Domain Object, just as we're doing now. So I'm changing to `schoology_extract_facade`. Note "extract" instead of "extractor"